src/main/utils/shell-env/compute-unix-shell-environment.injectable.ts

```diff
@@ -7,96 +7,149 @@ import { getInjectable } from "@ogre-tools/injectable";
 import getBasenameOfPathInjectable from "../../../common/path/get-basename.injectable";
 import spawnInjectable from "../../child-process/spawn.injectable";
 import randomUUIDInjectable from "../../crypto/random-uuid.injectable";
+import loggerInjectable from "../../../common/logger.injectable";
+import processExecPathInjectable from "./execPath.injectable";
+import processEnvInjectable from "./env.injectable";
+import { object } from "../../../common/utils";
+import type { AsyncResult } from "../../../common/utils/async-result";
 
 export interface UnixShellEnvOptions {
   signal: AbortSignal;
 }
 
-export type ComputeUnixShellEnvironment = (shell: string, opts: UnixShellEnvOptions) => Promise<EnvironmentVariables>;
+export type ComputeUnixShellEnvironment = (shell: string, opts: UnixShellEnvOptions) => Promise<AsyncResult<EnvironmentVariables, string>>;
+
+const getResetProcessEnv = (src: Partial<Record<string, string>>, overrides: Partial<Record<string, string>>): {
+  resetEnvPairs: (target: Partial<Record<string, string>>) => void;
+  env: Partial<Record<string, string>>;
+} => {
+  const originals = object.entries(overrides).map(([name]) => [name, src[name]] as const);
+
+  return {
+    env: {
+      ...src,
+      ...overrides,
+    },
+    resetEnvPairs: (target) => {
+      for (const [name, orginalValue] of originals) {
+        if (orginalValue) {
```

Comment:
I wonder if we should check something like `if (originalValue !== undefined)` to handle falsy values correctly?

Reply:
I guess `""` is falsy too, yeah can change that.